src/nlp/metric.py

```diff
@@ -212,6 +212,7 @@ def add(self, prediction=None, reference=None, **kwargs):
         """ Add one prediction and reference for the metric's stack.
         """
         example = {"predictions": prediction, "references": reference}
+        example = self.info.features.encode_example(example)
```

Comment:
Probably we'll want to move the info attributes one-level down in `Metrics` metrics like we did for `Dataset`.